### PR TITLE
doc/devguide: Installing Suricata from Git

### PR DESCRIPTION
**Pull request by mrbazzan**
As part of the process of moving documentation from redmine to "Read the Docs", this commit moves installing Suricata using git from redmine wiki into Suricata Developer Guide section. It also updates the necessary steps.

Signed-off-by: Bazzan <donbazzan@gmail.com>

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#5585](https://redmine.openinfosecfoundation.org/issues/5585)

Describe changes:
- Move redmine documentation to Read the Docs.
- Create `installation-from-git.rst` file in `userguide/devguide/codebase`
-  Update the ways to install `suricata-update`

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by mrbazzan** [comment]
Talk about "Basic Setup"
./convert.py file
ask if i should include other documentation in installation_from_git

While working on this, I noticed a couple of things (and I have a couple of questions):
- Is there any plan to move `suricata.git/doc/userguide/convert.py` to Python3?
- This PR only moves the "Ubuntu installation from Git" section, should I move the other pages in [installation-from-git](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Installation_from_GIT)

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+would do the regular "make install" and then it would automatically download
+and set up the latest ruleset from Emerging Threats available for Suricata
+
+.. code-block:: bash
+
+  ./configure && make && make install-full
+
+*make install-full*
+would combine everything mentioned above (install-conf and install-rules) -
+and will present you with a ready to run (configured and set up) Suricata
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/installation-from-git.rst:148]: nits:
set-up
add a period at the end of the sentence

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+
+Please continue with
+`Basic Setup <https://redmine.openinfosecfoundation.org/projects/suricata/
+wiki/Basic_Setup>`_.
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/installation-from-git.rst:153]: Let's replace this link with the one we have on readthedocs:

https://suricata.readthedocs.io/en/latest/quickstart.html#basic-setup

feel free to create an explicit target for the Basic setup section in the quickstart file, so you can use `ref` :)

(If you are unfamiliar with explicit targets for Sphinx, I find this useful:  https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#explicit-targets)

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+.. code-block:: bash
+
+  ./configure && make && make install-full
+
+*make install-full*
+would combine everything mentioned above (install-conf and install-rules) -
+and will present you with a ready to run (configured and set up) Suricata
+
+
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/installation-from-git.rst:150]: Let's add a subsection `Post installation` here, so this last couple of info is more contextualize :)

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+Please continue with
+`Basic Setup <https://redmine.openinfosecfoundation.org/projects/suricata/
+wiki/Basic_Setup>`_.
+
+In case you have already made a map for the most recent code, download the
+code into that map, and want to download recent code again, please enter:
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/installation-from-git.rst:156]: I find this portion rather difficult to understand. I think we can reword it to something like:

In case you have already created your Suricata directory and cloned the repository if you want to update your local repository with the most recent code, enter the Suricata directory and run:

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+  cd suricata/suricata
+
+next, enter:
+
+.. code-block:: bash
+
+  git pull
+
+After that, you start again at running *./autogen.sh*.
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/installation-from-git.rst:168]: Let's also reword this:

After that, you should run ``./autogen.sh`` again.

**Review by jufajardini — CHANGES_REQUESTED** [review]
Thanks for this contribution! :)

I have added some comments, but nothing too big to be changed.

With regards to the commit message, please adjust it so it conforms to our guidelines:
- we use the author form 'Name Surname'
- use an imperative verb in the commit message that helps one understand what your commit does (example: `devguide: add page on installing suricata from git`)
- there's no need to sign-off the commit message, as we keep you as the author, when we merge your PR ;)

**Comment by jufajardini** [comment]
> While working on this, I noticed a couple of things (and I have a couple of questions):
> 
> * Is there any plan to move `suricata.git/doc/userguide/convert.py` to Python3?
> * This PR only moves the "Ubuntu installation from Git" section, should I move the other pages in [installation-from-git](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Installation_from_GIT)

Thanks for asking :)
> * Is there any plan to move `suricata.git/doc/userguide/convert.py` to Python3?

I think that @jasonish may be a better person to answer this one

> * This PR only moves the "Ubuntu installation from Git" section, should I move the other pages in [installation-from-git](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Installation_from_GIT)

I _think_ that Ubuntu installation would be enough, but let's wait for Jason's word on this one, too :) In any case, that shouldn't impact the work done here.

**Comment by jufajardini** [comment]
About the author change, you can do that by running:
```bash
git config -g user.email donbazzan@gmail.com
git commit --amend --author "Don Bazzan <donbazzan@gmail.com>"
```
(Apologies if I got your name order wrong!)

**Comment by jufajardini** [comment]
About the author change, you can do that by running:
```bash
git config -g user.email donbazzan@gmail.com
git commit --amend --author "Don Bazzan <donbazzan@gmail.com>"
```
(Apologies if I got your name order wrong!)

**Comment by mrbazzan** [comment]
Thanks for the quick review. I'll be sure to update as soon as possible.

File: doc/userguide/devguide/codebase/installation-from-git.rst (hunk 5 above)
Comment (mrbazzan) [reply doc/userguide/devguide/codebase/installation-from-git.rst:168]: Just so I'm sure, is running `./autogen.sh` the only required step after pulling new changes? I thought there is a need to run `./configure`, `make` and other required command to re-install Suri. I'm not sure the reworded sentence shows this (unless I'm mistaken — English :( )

**Comment by mrbazzan** [comment]
Thanks for the quick review. I'll be sure to update as soon as possible.
> 
> (Apologies if I got your name order wrong!)

No problem. 👍🏿

File: doc/userguide/devguide/codebase/installation-from-git.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
+Installation from GIT
+=====================
+
+Ubuntu Installation from GIT
+----------------------------
+
+This document will explain how to install and use the most recent code of
+**Suricata** on Ubuntu. Installing from GIT or other operating systems is
```
[the remark this answers is not in the document]
Comment (mrbazzan) [reply doc/userguide/devguide/codebase/installation-from-git.rst:8]: Should I also remove the bold format for Suricata here?

File: doc/userguide/devguide/codebase/installation-from-git.rst (hunk 2 above)
Comment (mrbazzan) [reply doc/userguide/devguide/codebase/installation-from-git.rst:153]: I had no idea we had `basic-setup` on the readthedocs already (I should've checked, my bad)

Thanks for the cross-referencing link. 👍🏿

File: doc/userguide/devguide/codebase/installation-from-git.rst (hunk 5 above)
Comment (mrbazzan) [reply doc/userguide/devguide/codebase/installation-from-git.rst:168]: Oh...disregard my comment. There is an echo statement detailing the next step in `./autogen.sh`

File: doc/userguide/devguide/codebase/installation-from-git.rst (hunk 2 above)
Comment (jufajardini) [reply doc/userguide/devguide/codebase/installation-from-git.rst:153]: It's no problems, it's my duty to double-check these things ;)

File: doc/userguide/devguide/codebase/installation-from-git.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/devguide/codebase/installation-from-git.rst:8]: yes, please :)

**Comment by jasonish** [comment]
> * Is there any plan to move `suricata.git/doc/userguide/convert.py` to Python3?

I don't think so.  I had thought this script was done with and ready to be moved.  I suppose if it still has use, it could be moved to Python 3.. I wonder if a simple 2to3 script would do it?

**Comment by jasonish** [comment]
> I _think_ that Ubuntu installation would be enough, but let's wait for Jason's word on this one, too :) In any case, that shouldn't impact the work done here.

Probably should.  I don't think these docs should be moved blindly.  They should be verified and updated to make sure they are current.

**Comment by jufajardini** [comment]
> > I _think_ that Ubuntu installation would be enough, but let's wait for Jason's word on this one, too :) In any case, that shouldn't impact the work done here.
> 
> Probably should. I don't think these docs should be moved blindly. They should be verified and updated to make sure they are current.

Wise mentors are wise :D thanks 🙇🏽 
In that case, it would be best to leave that one for someone who can test the process with OpenBSD, then, right?
I'll create a ticket for that.

**Comment by mrbazzan** [comment]
> > * Is there any plan to move `suricata.git/doc/userguide/convert.py` to Python3?
> 
> I don't think so. I had thought this script was done with and ready to be moved. I suppose if it still has use, it could be moved to Python 3.. I wonder if a simple 2to3 script would do it?

**Comment by jufajardini** [comment]
> > I don't think so. I had thought this script was done with and ready to be moved.
> 
> I used it when creating `installation-from-git.rst` file, it was quite useful although I had to edit some lines
> 
> > I suppose if it still has use, it could be moved to Python 3.. I wonder if a simple 2to3 script would do it?
> 
> A simple 2to3 script and a couple of edits made it work. @jufajardini can I get an issue for the conversion?

Sorry for the late answer, I've created the ticket and assigned it to yourself: https://redmine.openinfosecfoundation.org/issues/5596 :)

**Comment by mrbazzan** [comment]
> 
> Sorry for the late answer, I've created the ticket and assigned it to yourself: https://redmine.openinfosecfoundation.org/issues/5596 :)

Thank you. I'll get to it.
